rasa_nlu/registry.py

```diff
@@ -0,0 +1,76 @@
+"""This is a somewhat delicate package. It contains all registered components and preconfigured templates.
+
+Hence, it imports all of the components. To avoid cycles, no component should import this in module scope."""
+from typing import Optional
+from typing import Type
+
+from rasa_nlu.classifiers.keyword_intent_classifier import KeywordIntentClassifier
+from rasa_nlu.classifiers.mitie_intent_classifier import MitieIntentClassifier
+from rasa_nlu.classifiers.sklearn_intent_classifier import SklearnIntentClassifier
+from rasa_nlu.extractors.entity_synonyms import EntitySynonymMapper
+from rasa_nlu.extractors.mitie_entity_extractor import MitieEntityExtractor
+from rasa_nlu.extractors.spacy_entity_extractor import SpacyEntityExtractor
+from rasa_nlu.featurizers.mitie_featurizer import MitieFeaturizer
+from rasa_nlu.featurizers.ngram_featurizer import NGramFeaturizer
+from rasa_nlu.featurizers.spacy_featurizer import SpacyFeaturizer
+from rasa_nlu.tokenizers.mitie_tokenizer import MitieTokenizer
+from rasa_nlu.utils.mitie_utils import MitieNLP
+from rasa_nlu.utils.spacy_utils import SpacyNLP
+
+# Classes of all known components. If a new component should be added, its class needs to be listed here.
+component_classes = [
+    SpacyNLP, SpacyEntityExtractor, SklearnIntentClassifier, SpacyFeaturizer,
+    MitieNLP, MitieEntityExtractor, MitieIntentClassifier, MitieFeaturizer, MitieTokenizer,
+    KeywordIntentClassifier, EntitySynonymMapper, NGramFeaturizer]
+
+# Mapping from a components name to its class to allow name based lookup.
+registered_components = {component.name: component for component in component_classes}
+
+# To simplify usage, there are a couple of model templates, that already add necessary components in the right order.
+# They also implement the preexisting `backends`.
+registered_model_templates = {
```

Comment:
yep cool - I like that. So we can have a 'named' pipeline which is just the name of a template, or if you pass a list then it's interpreted as a custom pipeline.